src/bdecode.cpp

```diff
@@ -289,6 +289,93 @@ namespace {
 		m_buffer = buf;
 	}
 
+	bool bdecode_node::has_soft_error(span<char> error) const
+	{
+		bdecode_token const* tokens = m_root_tokens;
+		int token = m_token_idx;
+
+		// we don't know what the original depth_limit was
+		// so this has to go on the heap
+		std::vector<int> stack;
+		// make the initial allocation the default depth_limit
+		stack.reserve(100);
+
+		do
+		{
+			switch (tokens[token].type)
```

Comment:
this just occurred to me; is it possible to have 0 tokens? i.e. have this be an out-of-bound access?